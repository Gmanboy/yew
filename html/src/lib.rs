use proc_macro_hack::proc_macro_hack;

/// Generate html tree
#[proc_macro_hack]
pub use yew_html_impl::html;
