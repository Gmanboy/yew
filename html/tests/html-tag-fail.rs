use yew_html::html;

fn main() {
    html! {
        <div>
    };

    html! {
        <div><div>
    };

    html! {
        </div>
    };

    html! {
        <div><div></div>
    };

    html! {
        <div></div>
        <div></div>
    };

    html! {
        <div></span>
    };
}
