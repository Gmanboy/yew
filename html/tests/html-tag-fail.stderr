error: this open tag has no corresponding close tag
 --> $DIR/html-tag-fail.rs:5:9
  |
5 |         <div>
  |         ^^^^^

error: this open tag has no corresponding close tag
 --> $DIR/html-tag-fail.rs:9:9
  |
9 |         <div><div>
  |         ^^^^^

error: expected valid html element
  --> $DIR/html-tag-fail.rs:13:9
   |
13 |         </div>
   |         ^

error: this open tag has no corresponding close tag
  --> $DIR/html-tag-fail.rs:17:9
   |
17 |         <div><div></div>
   |         ^^^^^

error: only one root html element allowed
  --> $DIR/html-tag-fail.rs:22:9
   |
22 |         <div></div>
   |         ^

error: this open tag has no corresponding close tag
  --> $DIR/html-tag-fail.rs:26:9
   |
26 |         <div></span>
   |         ^^^^^
